src/YesSql.Core/Sql/SqlBuilder.cs

```diff
@@ -71,10 +71,10 @@ public void Skip(string skip)
         public void Take(string take)
             => _count = take;
 
-        public virtual void Join(JoinType type, string table, string onTable, string onColumn, string toTable, string toColumn, string schema, string alias = null, string toAlias = null)
+        public virtual void Join(JoinType type, string table, string onTable, string onColumn, string toTable, string toColumn, string schema, string alias = null, string toAlias = null, bool onTableIsAlias = false)
         {
             // Don't prefix if alias is used
-            if (alias != onTable)
+            if (alias != onTable && !onTableIsAlias)
```

Comment:
Seeing this, I believe we don't need a new field and just to set the `alias` argument instead such that it knows that `onTable` is an alias.